src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -949,11 +969,11 @@ void Replayer<I>::create_mirror_snapshot(
     {}, cls::rbd::GROUP_SNAPSHOT_STATE_INCOMPLETE};
   local_snap.name = prepare_non_primary_mirror_snap_name(m_global_group_id,
       group_snap_id);
-  m_local_group_snaps.push_back(local_snap);
+  m_create_group_snaps.push_back(local_snap);
```

Comment:
Instead of adding an entry to the vector before actually creating it on disk, I would suggest creating it and then relisting the snaps .